javaslang/src/main/java/javaslang/collection/Tree.java

```diff
@@ -729,6 +751,7 @@ public Node(T value, List<Node<T>> children) {
             Objects.requireNonNull(children, "children is null");
             this.value = value;
             this.children = children;
+            this.size = children.size() + 1;
```

Comment:
/**
     * Returns the number of nodes (including root and leafs).
     *
     * @return The size of the tree.
     */

So this should be applied recursively?